OpenSSL version check fails

### Steps to reproduce the behavior

Ubuntu 22.04 run `CONFIGURE_OPTS="--enable-yjit" /usr/local/bin/ruby-build 3.2.2 /usr/local`

Output:
```
To follow progress, use 'tail -f /tmp/ruby-build.20231104112055.112604.log' or pass --verbose
/usr/local/bin/ruby-build: line 1062: "3: syntax error: operand expected (error token is ""3")
Downloading openssl-3.1.4.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/840af5366ab9b522bde525826be3ef0fb0af81c6a9ebd84caa600fea1731eee3
Installing openssl-3.1.4...

BUILD FAILED (Ubuntu 22.04 using ruby-build 20231025)

Inspect or clean up the working tree at /tmp/ruby-build.20231104112055.112604.rX0bQA
Results logged to /tmp/ruby-build.20231104112055.112604.log
Last 10 log lines:
make[1]: Entering directory '/tmp/ruby-build.20231104112055.112604.rX0bQA/openssl-3.1.4'
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --enable-yjit -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/openssl/ssl\"" -DENGINESDIR="\"/usr/local/openssl/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/openssl/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DZLIB -DZLIB_SHARED -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_libctx.d.tmp -MT apps/lib/libapps-lib-app_libctx.o -c -o apps/lib/libapps-lib-app_libctx.o apps/lib/app_libctx.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --enable-yjit -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/openssl/ssl\"" -DENGINESDIR="\"/usr/local/openssl/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/openssl/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DZLIB -DZLIB_SHARED -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_params.d.tmp -MT apps/lib/libapps-lib-app_params.o -c -o apps/lib/libapps-lib-app_params.o apps/lib/app_params.c
cc1: error: unknown pass yjit specified in '-fenable'
cc1: error: unknown pass yjit specified in '-fenable'
make[1]: *** [Makefile:4295: apps/lib/libapps-lib-app_params.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:4287: apps/lib/libapps-lib-app_libctx.o] Error 1
make[1]: Leaving directory '/tmp/ruby-build.20231104112055.112604.rX0bQA/openssl-3.1.4'
make: *** [Makefile:3253: build_sw] Error 2
```
### Expected vs. actual behavior

Working install vs broken install

### Reason

- the OpenSSL version check changed (and now fails)
- the script tries to download and install OpenSSL because it thinks it doesn't have an acceptable version installed
- the openssl compilation uses the `CONFIGURE_OPTS` that were meant for the Ruby compilation

So although the third bullet is an issue by itself, it all starts with the version check.

/usr/local/bin/ruby-build: line 1062: "3: syntax error: operand expected (error token is ""3")

Digging into this, the version check was [changed ](https://github.com/rbenv/ruby-build/commit/31e53468b65fd3f881bd4ba07b5f0131e1663971#diff-9161f3dbce993baa919f6535fbe4c32d2d2cd1ba9e5c93aa899188063c9022acR1034)in commit `31e5346`  and since then it's broken (for me).

The new version check returns a quote before the version number and that causes `normalize_semver` to choke on it.

```
root@teststable:/usr/local/bin# cc -xc -E - <<EOF 2>/dev/null | sed -n 's/OpenSSL \([0-9][0-9.]*\).*/\1/p'
#include <openssl/opensslv.h>
OPENSSL_VERSION_TEXT
EOF
"3.0.2
root@teststable:/usr/local/bin#
```

Digging deeper:
```
root@teststable:/usr/local/bin# cc -xc -E - <<EOF 2>/dev/null
> #include <openssl/opensslv.h>
> OPENSSL_VERSION_TEXT
> EOF
# 0 "<stdin>"
# 0 "<built-in>"
# 0 "<command-line>"
# 1 "/usr/include/stdc-predef.h" 1 3 4
# 0 "<command-line>" 2
# 1 "<stdin>"
# 1 "/usr/include/openssl/opensslv.h" 1 3 4
# 15 "/usr/include/openssl/opensslv.h" 3 4
        
# 109 "/usr/include/openssl/opensslv.h" 3 4
# 1 "/usr/include/openssl/macros.h" 1 3 4
# 12 "/usr/include/openssl/macros.h" 3 4
        

# 1 "/usr/include/x86_64-linux-gnu/openssl/opensslconf.h" 1 3 4
# 12 "/usr/include/x86_64-linux-gnu/openssl/opensslconf.h" 3 4
        

# 1 "/usr/include/x86_64-linux-gnu/openssl/configuration.h" 1 3 4
# 15 "/usr/include/x86_64-linux-gnu/openssl/configuration.h" 3 4
        
# 15 "/usr/include/x86_64-linux-gnu/openssl/opensslconf.h" 2 3 4
# 15 "/usr/include/openssl/macros.h" 2 3 4
# 110 "/usr/include/openssl/opensslv.h" 2 3 4
# 2 "<stdin>" 2

# 2 "<stdin>" 3 4
"OpenSSL 3.0.2 15 Mar 2022"

root@teststable:/usr/local/bin# echo '"OpenSSL 3.0.2 15 Mar 2022"'| sed -n 's/OpenSSL \([0-9][0-9.]*\).*/\1/p'
"3.0.2
```

The previous version check does not have that problem.

--- a/bin/ruby-build
+++ b/bin/ruby-build
@@ -1034,7 +1034,7 @@ has_broken_mac_openssl() {
 
 # Detect the OpenSSL version that a compiler can reasonably link to.
 system_openssl_version() {
-  cc -xc -E - <<EOF 2>/dev/null | sed -n 's/OpenSSL \([0-9][0-9.]*\).*/\1/p'
+  cc -xc -E - <<EOF 2>/dev/null | sed -n 's/"\{0,1\}OpenSSL \([0-9][0-9.]*\).*/\1/p'
 #include <openssl/opensslv.h>
 OPENSSL_VERSION_TEXT
 EOF

--- a/test/build.bats
+++ b/test/build.bats
@@ -322,7 +322,7 @@ OUT
   stub_repeated uname '-s : echo Linux'
   stub_repeated brew false
   # shellcheck disable=SC2016
-  stub cc '-xc -E - : [[ "$(cat)" == *OPENSSL_VERSION_TEXT* ]] && printf "# <unrelated> 4.0.2\nOpenSSL 1.0.3a  1 Aug 202\n0 errors.\n"'
+  stub cc '-xc -E - : [[ "$(cat)" == *OPENSSL_VERSION_TEXT* ]] && printf "# <unrelated> 4.0.2\n\"OpenSSL 1.0.3a  1 Aug 202\"\n0 errors.\n"'
   stub_make_install
 
   mkdir -p "$INSTALL_ROOT"/openssl/ssl # OPENSSLDIR

